components/catalogs/front/src/components/ProductSelection/criteria/CategoryCriterion/CategoryCriterion.tsx

```diff
@@ -0,0 +1,73 @@
+import React, {FC, useEffect, useState} from 'react';
+import styled from 'styled-components';
+import {CloseIcon, IconButton, List} from 'akeneo-design-system';
+import {Operator} from '../../models/Operator';
+import {CriterionModule} from '../../models/Criterion';
+import {CategoryCriterionState} from './types';
+import {useTranslate} from '@akeneo-pim-community/shared';
+import {CategoryOperatorInput} from './CategoryOperatorInput';
+import {CategorySelectInput} from './CategorySelectInput';
+import {ErrorHelpers} from '../ErrorHelpers';
+
+const Fields = styled.div`
+    display: flex;
+    gap: 20px;
+    flex-wrap: wrap;
+    flex-grow: 1;
+`;
+
+const Field = styled.div`
+    flex-basis: 200px;
+    flex-shrink: 0;
+`;
+
+const LargeField = styled.div`
+    flex-basis: 300px;
+    flex-shrink: 0;
+`;
+
+const CategoryCriterion: FC<CriterionModule<CategoryCriterionState>> = ({state, errors, onChange, onRemove}) => {
+    const translate = useTranslate();
+    const hasError = Object.values(errors).filter(n => n).length > 0;
+    const [showCategories, setShowCategories] = useState<boolean>(false);
+
+    useEffect(() => {
+        setShowCategories(Operator.UNCLASSIFIED !== state.operator);
+    }, [state.operator]);
```

Comment:
the useEffect is useless here.
To understand more where useEffect should be used and why, and understand why this particular usage is not recommended, you can checkout the new react documentation about it.
https://beta.reactjs.org/learn/you-might-not-need-an-effect#updating-state-based-on-props-or-state

```suggestion
    const showCategories = Operator.UNCLASSIFIED !== state.operator;
```

Yes, the same mistake was done in FamilyCriterion but I already planned to fix it, see https://github.com/akeneo/pim-community-dev/pull/17694/files